Client/script.js

```diff
@@ -0,0 +1,45 @@
+//Event Listeners
+
+const searchButton = document.getElementById('search')
+const randomButton = document.getElementById('randombutton')
+
+function showResults(e){
+    e.preventDefault();
+    let submitter = e.submitter
+    if (topic != ""){
+        fetch(`http://localhost:8080/topics/${topic}`)
+        .then(resp => resp.text())
+        .then(data => {
+
+        })
+    }
+}
+
+
+
+
+//for js file
+// const searchButton = document.querySelector("#search");
+// searchButton.addEventListener("click", () => {
+//   fetch(`http://localhost:8080/foods`)
+//     .then((res) => res.text())
+//     .then(insertResults);
+//   window.location.href = "results.html";
```

Comment:
Are you here trying to change the dom to be of the results page and not the index? i think window.location.href just takes you to the page specified. Perhaps you can try and put your data in sessionstorage to send across to the other document? https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage